.github/workflows/test-reporting.yml

```diff
@@ -0,0 +1,65 @@
+---
+name: Generate junit test report
+
+on:  # yamllint disable-line rule:truthy
+  pull_request:  # yamllint disable-line rule:empty-values
+  push:
+    branches: [devel]
+
+env:
+  DESIRED_GO_VERSION: '1.20'
+
+jobs:
+  go_test_coverage:
+    name: go test coverage
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.DESIRED_GO_VERSION }}
+
+      - name: build and install receptor
+        run: |
+          make build-all
+          sudo cp ./receptor /usr/local/bin/receptor
+
+      - name: Download kind binary
+        run: curl -Lo ./kind https://kind.sigs.k8s.io/dl/latest/kind-linux-amd64 && chmod +x ./kind
+
+      - name: Create k8s cluster
+        run: ./kind create cluster
+
+      - name: Interact with the cluster
+        run: kubectl get nodes
+
+      - name: Run receptor tests with coverage
```

Comment:
Maybe something like this: 
- name: Install go junit reporting
        run: make go-junit-report

      - name: Run receptor tests
        run: PATH="${PWD}:${PATH}" go test -v 2>&1 ./... | go-junit-report -set-exit-code > report.xml